src/components/calendar/bl-calendar.ts

```diff
@@ -0,0 +1,133 @@
+import { CSSResultGroup, html, LitElement } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+import "../button/bl-button";
+import "../icon/bl-icon";
+import { DAYS, MONTHS } from "./bl-calendar.constant";
+import style from "./bl-calendar.css";
+import { TCalendarTypes, TMonth } from "./bl-calendar.types";
+
+@customElement("bl-calendar")
+export default class BlCalendar extends LitElement {
+  @property()
+  type: TCalendarTypes = "single";
+
+  @property()
+  minDate?: Date | string;
+
+  @property()
+  maxDate?: Date | string;
+
+  @property({ type: Array, attribute: "default-value", reflect: true })
+  defaultValue?: Date | Date[];
+
+  @property({ type: Number, attribute: "start-of-week", reflect: true })
+  startOfWeek?: number = 1;
+
+  @property()
+  locale?: string = document.documentElement.lang;
+
+  @property({ type: Array })
+  disabledDates?: Date | Date[];
+
+  @state()
+  private day: number;
+
+  @state()
+  private month: TMonth = MONTHS[new Date().getMonth()];
+
+  @state()
+  private year: number = new Date().getFullYear();
+
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+  render() {
+    this.getWeekDayOfDate();
+    const calendarDays = this.createCalendarDays();
+
+    return html`<div>
+      <div class="calendar-content">
+        <div class="selected-month-year-content">
+          <bl-button
+            class="arrows"
+            icon="arrow_left"
+            variant="tertiary"
+            kind="neutral"
+            @click="${() => this.setPreviousMonth()}"
+          ></bl-button>
+          <div class="selected-month-year">
+            <span class="day-text">${this.month.name}</span>
+            <span class="day-text">${this.year}</span>
+          </div>
+          <bl-button
+            class="arrows"
+            icon="arrow_right"
+            variant="tertiary"
+            kind="neutral"
+            @click="${() => this.setNextMonth()}"
+          ></bl-button>
+        </div>
+        <div class="weekdays">
+          ${Array.from(calendarDays).map(([key, value]) => {
+            return html`<div class="calendar-column">
+              <div class="weekday-text calendar-cell">${DAYS[key]}</div>
+              ${value.map(day => {
+                return html` <div class="day-text calendar-cell">${day}</div>`;
+              })}
+            </div>`;
+          })}
+        </div>
+      </div>
+    </div> `;
+  }
+  setMonth(monthIndex: number) {
+    this.month = MONTHS[monthIndex];
+  }
+
+  setYear(year: number) {
+    this.year = year;
+  }
+
+  createCalendarDays() {
+    const monthlyCalendar: Map<number, (number | string)[]> = new Map();
+    let i = 0;
+    let dayOfTheWeek = 0;
+    let iteratedDay = 1;
+
+    for (; i < this.getDayNumInAMonth() + this.getWeekDayOfDate(); i += 1) {
```

Comment:
I believe using the map or forEach functions instead of a basic for loop would enhance readability and performance.